packages/connection-info/src/connection-secrets.ts

```diff
@@ -190,10 +181,75 @@ export function extractSecrets(connectionInfo: Readonly<ConnectionInfo>): {
   return { connectionInfo: connectionInfoWithoutSecrets, secrets };
 }
 
-function omitPropertiesWhoseValuesAreEmptyObjects<
-  T extends Record<string, Record<string, unknown>>
->(obj: T): Partial<T> {
+function omitPropertiesWhoseValuesAreEmptyObjects<T extends Document>(obj: T) {
   return Object.fromEntries(
     Object.entries(obj).filter(([, value]) => Object.keys(value).length > 0)
-  ) as Partial<T>;
+  );
+}
+
+function extractAutoEncryptionSecrets(data: Document) {
```

Comment:
```suggestion
function extractAutoEncryptionSecrets(data: AutoEncryptionOptions) {
```

and this should have a return type annotation as well